lib/minimax.rb

```diff
@@ -0,0 +1,49 @@
+class Minimax
+
+  def find_best_move(board, depth=0, best_score={}, current_player, player1, player2)
+    opponent = set_opponent(current_player, player1, player2)
+    available_squares = board.available_squares
+
+    return score_move(board, depth, current_player, opponent) if 
+      board.stop_playing?(player1, player2)
+
+    available_squares.each do |square|
+      copy_of_board = board.copy_board
+      board.player_make_move(current_player, square)
```

Comment:
I can't quite work out why there is a `copy_of_board` but that moves and resets are made on the `board`?

Reply:
Thanks Ashley, I totally missed this! I'll change it so `copy_of_board` is used within the .each iteration